libmount/src/tab.c

```diff
@@ -414,15 +438,71 @@ int mnt_table_add_fs(struct libmnt_table *tb, struct libmnt_fs *fs)
 	if (!list_empty(&fs->ents))
 		return -EBUSY;
 
-	mnt_ref_fs(fs);
-	list_add_tail(&fs->ents, &tb->ents);
-	tb->nents++;
+	if (!mnt_table_find_fs(tb, fs)) {
+		mnt_ref_fs(fs);
+		list_add_tail(&fs->ents, &tb->ents);
+		tb->nents++;
+	}
 
```

Comment:
This seems unnecessary as the fs->ents has to be empty and fs cannot be member of any table. 

(I'm going to fix it before merge.)